sonic_y_cable/y_cable_base.py

```diff
@@ -0,0 +1,589 @@
+        """
+
+        raise NotImplementedError
+
+    def toggle_mux_to_torB(self, port):
+        """
+        This API specifically does a hard switch toggle of the Y cable's MUX regardless of link state to
+        TOR B. This means if the Y cable is actively routing, the "check_active_linked_tor_side"
+        API will now return Tor B. It also implies that if the link is actively routing on this port, Y cable
+        MUX will start forwarding packets from TOR B to NIC, and drop packets from TOR A to NIC
+        regardless of previous forwarding state.
+
+        Args:
+            port:
+                an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+        Returns:
+            a Boolean, True if the toggle succeeded and False if it did not succeed.
+        """
+        raise NotImplementedError
+
+    def check_read_side(self, port):
+        """
+        This API specifically checks which side of the Y cable the reads are actually getting performed
+        from, either TOR A or TOR B or NIC and returns the value.
+
+        Args:
+            port:
+                an Integer, the actual physical port connected to Y end of a Y cable which can which side reading the MUX from
+
+        Returns:
+            One of the following predefined constants:
+                TARGET_TORA, if reading the Y cable from TOR A side.
+                TARGET_TORB, if reading the Y cable from TOR B side.
+                TARGET_NIC, if reading the Y cable from NIC side.
+                TARGET_UNKNOWN, if reading the Y cable API fails.
+        """
+
+        raise NotImplementedError
+
+    def check_mux_direction(self, port):
+        """
+        This API specifically checks which side of the Y cable mux is currently point to
+        and returns either TOR A or TOR B. Note that this API should return mux-direction
+        regardless of whether the link is active/routing state or not.
+
+        Args:
+            port:
+                an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            One of the following predefined constants:
+                TARGET_TORA, if mux is pointing to TOR A side.
+                TARGET_TORB, if mux is pointing to TOR B side.
+                TARGET_UNKNOWN, if mux direction API fails.
+        """
+
+        raise NotImplementedError
+
+    def check_active_linked_tor_side(self, port):
+        """
+        This API specifically checks which side of the Y cable is actively linked and routing/sending
+        and returns either TOR A or TOR B.
+
+        Args:
+            port:
+                an Integer, the actual physical port connected to a Y cable
+
+        Returns:
+            One of the following predefined constants:
+                TARGET_TORA, if TOR A is actively linked and routing/sending traffic.
+                TARGET_TORB, if TOR B is actively linked and routin/sending traffic.
```

Comment:
s/routin/routing/

Reply:
On second thought, it may be better to have it only as 'sending', as per https://github.com/Azure/sonic-platform-common/pull/186/files#r626912311